docs/core/advanced_tutorials/using-result-handlers.md

```diff
@@ -0,0 +1,133 @@
+
+## Setting up to handle results
+
+The first step to getting the most out of Result Handlers is to enable results storage. You must enable in two places: globally for the Prefect installation, and at the task level by providing a result handler to tasks either through its flow initialization or as a task-level override. Depending on your storage backend, you may also need to ensure your authentication credentials are set up properly. For Cloud customers, some of this is handled by default.
+
+For Core-only users or Cloud customers on Prefect versions <0.9.1, you must
+- opt-in to checkpointing globally by setting the `prefect.config.flows.checkpointing` to "True".
+- specify the result handler your tasks will use for at least one level of specificity (flow-level or task-level)
+
+For Cloud customers on Prefect version 0.9.1+,
+- checkpointing will automatically be turned on; you can disable it by setting `prefect.config.flows.checkpointing` to "False"
+ - a result handler that matches the storage backend of your `prefect.config.flows.storage` setting will automatically be applied to all tasks , if available; notably this is not yet supported for Docker Storage
+ - you can override the automatic results handler at the global level, flow level, or task level
+
+#### Setting result handler at the Flow level
+```bash
+export PREFECT__FLOWS__CHECKPOINTING=true
+```
+ ```python
+# flow.py
+
+@task
+def add(x, y=1):
+    return x + y
+
+# send the configuration to the Flow object
+with Flow("my handled flow!", result_handler=LocalStorageHandler()):
+    first_result = add(1, y=2)
+    second_result = add(x=first_result, y=100)
+```
+
+#### Setting result handler at the task level
+```bash
+export PREFECT__FLOWS__CHECKPOINTING=true
+```
+ ```python
+# flow.py
+
+# configure on the task decorator
+@task(result_handler=LocalStorageHandler())
+def add(x, y=1):
+    return x + y
+
+with Flow("my handled flow!"):
+    first_result = add(1, y=2)
+    # or send as a keyword argument at task initialization
+    second_result = add(x=first_result, y=100, result_handler=LocalStorageHandler())
+```
+
+## Choosing your result handlers
+
+In the above examples, we only used the `LocalStorageHandler` class. This is one of several result handlers that integrate with different storage backends; the full list is in the API docs for [prefect.engine.results_handler](../../api/latest/engine/result_handlers.html) and more details on this interface is described in the concept ["Results and Result Handlers"](../concepts/results.md) documentation.
+ 
+We can write our own result handlers as long as they extend the `ResultHandler` interface, or we can pick an existing implementation from Prefect Core that utilizes a storage backend we like; for example, I will use `prefect.engine.results_handler.GCSResultHandler` so that my data will be persisted in Google Cloud Storage.
```

Comment:
```suggestion
We can write our own result handlers as long as they extend the [`ResultHandler`](https://github.com/PrefectHQ/prefect/blob/master/src/prefect/engine/result_handlers/result_handler.py) interface, or we can pick an existing implementation from Prefect Core that utilizes a storage backend we like; for example, I will use `prefect.engine.results_handler.GCSResultHandler` so that my data will be persisted in Google Cloud Storage.
```

Reply:
For a link to the result handler interface :)